sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -37,14 +37,18 @@ object Klaviyo {
      */
     private val preInitQueue: Queue<Operation<Unit>> = LinkedList()
 
-    init {
-        /**
-         * Since the analytics module owns ApiClient, we must register it.
-         *
-         * This registration is a lambda invoked when the API service is required.
-         * KlaviyoApiClient service is not being initialized here.
-         */
-        Registry.registerOnce<ApiClient> { KlaviyoApiClient }
+    /**
+     * Since the analytics module owns these services, it must register them.
+     *
+     * This registration is a lambda invoked when the service is required, not instantiated now
+     */
+    private fun initializeServices() = Registry.apply {
+        registerOnce<ApiClient> { KlaviyoApiClient }
+        registerOnce<State> {
+            KlaviyoState().also { state ->
+                register<StateSideEffects>(StateSideEffects(state))
+            }
+        }
     }
```

Comment:
Previously I was using the one-time nature of an `init` (called the first time you access Klaviyo, usually to initialize) to have APIClient register only once. Now that we have `RegisterOnce`, I don't have to use an Object initializer, which I tend to avoid anyway.